ui/src/app/views/notebook-redirect/component.ts

```diff
@@ -61,7 +132,18 @@ export class NotebookRedirectComponent implements OnInit, OnDestroy {
   ngOnInit(): void {
     this.wsNamespace = this.route.snapshot.params['ns'];
     this.wsId = this.route.snapshot.params['wsid'];
-    this.notebookName = this.route.snapshot.params['nbName'];
+    this.creating = this.route.snapshot.data.creating;
+
+    if (this.creating) {
+      this.notebookName = this.route.snapshot.queryParamMap.get('notebook-name');
+      if (this.route.snapshot.queryParamMap.get('kernel-type') === Kernels.R.toString()) {
+        this.fileContent = JSON.stringify(rNotebookFileContent);
```

Comment:
nit: I think it would be cleaner to assign `this.kernelType` instead, then move this file content switch right before the putContents call. It feels to me like `kernelType` is a property of the component while `fileContent` is a downstream implementation detail to be isolated to a local variable if possible.